android/src/main/java/com/guichaguri/trackplayer/service/player/LocalPlayback.java

```diff
@@ -120,6 +118,12 @@ public void remove(List<Integer> indexes, Promise promise) {
         }
     }
 
+    @Override
+    public void move(int index, int newIndex, Promise promise) {
+        queue.add(newIndex, queue.remove(index));
+        source.moveMediaSource(index, newIndex, manager.getHandler(), Utils.toRunnable(promise));
+    }
+
```

Comment:
@Guichaguri, Are we allowing using `move()` on the current index. SwiftAudio does not allow this. I would have to dig deeper to understand why.

Also just wondering, does `queue.add(newIndex, queue.remove(index));` actually insert at the correct slot, if `newIndex` is greater than `index`? Trying to learn a little more Java as I go along here. Is there a reason you're not using swap() instead?

Reply:
If SwiftAudio doesn't allow it, we can make Android reject that too. It would actually make it more consistent with `remove()` which doesn't allow removing the current track either.

For some reason I didn't even thought about `swap()` when I was writing this, nor that removing an item would shift all following indexes

Reply:
I guess I have the same basic question about shuffle. Doesn’t shuffle potentially move the current track?

Reply:
You're right, we can make it not move at all and move all the other tracks
Or we can also make it move all following tracks, so that previous ones don't play again, which makes more sense